src/java/org/apache/cassandra/index/sai/disk/format/Version.java

```diff
@@ -0,0 +1,131 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.index.sai.disk.format;
+
+import java.util.List;
+
+import com.google.common.base.Objects;
+import com.google.common.collect.Lists;
+
+import org.apache.cassandra.index.sai.IndexContext;
+import org.apache.cassandra.index.sai.disk.v1.V1OnDiskFormat;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Format version of indexing component, denoted as [major][minor]. Same forward-compatibility rules apply as to
+ * {@link org.apache.cassandra.io.sstable.format.Version}.
+ */
+public class Version
+{
+    // Current version
+    public static final Version BA = new Version("ba", V1OnDiskFormat.instance, (c, i) -> defaultFileNameFormat(c, i, "ba"));
+
+    // These should be added in reverse order so that the latest version is used first. Version matching tests
+    // are more likely to match the latest version, so we want to test that one first.
+    public static final List<Version> ALL = Lists.newArrayList(BA);
+
+    public static final Version EARLIEST = BA;
+    // The latest version can be configured to be an earlier version to support partial upgrades that don't
+    // write newer versions of the on-disk formats.
+    public static final Version LATEST = parse(System.getProperty("cassandra.sai.latest.version", "ba"));
+
+    private final String version;
+    private final OnDiskFormat onDiskFormat;
+    private final FileNameFormatter fileNameFormatter;
+
+    private Version(String version, OnDiskFormat onDiskFormat, FileNameFormatter fileNameFormatter)
+    {
+        this.version = version;
+        this.onDiskFormat = onDiskFormat;
+        this.fileNameFormatter = fileNameFormatter;
+    }
+
+    public static Version parse(String input)
+    {
+        checkArgument(input != null);
+        checkArgument(input.length() == 2);
```

Comment:
nit: Might be worth just keeping a set above of usable versions and complaining w/ a nice error message when `contains()` fails?